documentation/specs/src/economics/fee-system.md

```diff
@@ -1,7 +1,378 @@
-## Fee system
+# Fee system
 
-In order to be accepted by the Namada ledger, transactions must pay fees. Transaction fees serve two purposes: first, the efficient allocation of block space given permissionless transaction submission and varying demand, and second, incentive-compatibility to encourage block producers to add transactions to the blocks which they create and publish.
+In order to be accepted by the Namada ledger, transactions must pay fees.
+Transaction fees serve two purposes: first, the efficient allocation of block
+space and gas (which are scarce resources) given permissionless transaction
+submission and varying demand, and second, incentive-compatibility to encourage
+block producers to add transactions to the blocks which they create and publish.
 
-Namada transaction fees can be paid in any fungible token which is a member of a whitelist controlled by Namada governance. Governance also sets minimum fee rates (which can be periodically updated so that they are usually sufficient) which transactions must pay in order to be accepted (but they can always pay more to encourage the proposer to prioritise them). When using the shielded pool, transactions can also unshield tokens in order to pay the required fees.
+Namada transaction fees can be paid in any fungible token which is a member of a
+whitelist controlled by Namada governance. Governance also sets minimum fee
+rates (which can be periodically updated so that they are usually sufficient)
+which transactions must pay in order to be accepted (but they can always pay
+more to encourage the proposer to prioritize them). When using the shielded
+pool, transactions can also unshield tokens in order to pay the required fees.
 
-The token whitelist consists of a list of $(T, GP_{min})$ pairs, where $T$ is a token identifier and $GP_{min}$ is the minimum price per unit gas which must be paid by a transaction paying fees using that asset. This whitelist can be updated with a standard governance proposal. All fees collected are paid directly to the block proposer (incentive-compatible, so that side payments are no more profitable).
\ No newline at end of file
+The token whitelist consists of a list of $(T, GP_{min})$ pairs, where $T$ is a
+token identifier and $GP_{min}$ is the minimum (base) price per unit gas which
+must be paid by a transaction paying fees using that asset. This whitelist can
+be updated with a standard governance proposal. All fees collected are paid
+directly to the block proposer (incentive-compatible, so that side payments are
+no more profitable).
+
+Fees are distributed among the delegators with the mechanism explained in the
```

Comment:
No, they aren't - fees go only to the proposer. Proof-of-stake _rewards_ are distributed to the delegators.

(otherwise it's profitable for the proposer to accept side payments instead)

Reply:
ah as we have a `GP_{min}` we could actually pay `gas * GP_min` to everyone if we wanted, and the difference to the proposer, that would also be incentive-compatible

Reply:
do you think that this would be easy to do? it is slightly preferable if possible

Reply:
It is technically feasible but very inefficient. I think it would be better to keep it as is for the moment and then implement the distribution once #1115 has been completed

Reply:
wait why would this be inefficient?

Reply:
Because at the moment we don't have an easy way to get all the delegators for a specific validator. To do so we need to iterate over all the bonds deltas

Reply:
Ah, but staking fee distribution should be able to accomplish this (I don't suggest iterating). We can save this for a future upgrade, though.